vllm/utils.py

```diff
@@ -264,7 +268,10 @@ def get_ip() -> str:
     return "0.0.0.0"
 
 
-def get_distributed_init_method(ip: str, port: int) -> str:
+@lru_cache(maxsize=None)
+def get_distributed_init_method() -> str:
+    ip = get_ip()
+    port = get_open_port()
```

Comment:
is it safe to cache here? what if we want to init distributed for two different groups, and we expect the function to return different results per call?